packages/api/src/interfaces/scripts/ScdlDataIntegration.node.test.ts

```diff
@@ -0,0 +1,291 @@
+        addProducerMock = jest.spyOn(ScdlCli.prototype, "addProducer").mockResolvedValue();
+        parseMock = jest.spyOn(ScdlCli.prototype, "parse").mockResolvedValue();
+        parseXlsMock = jest.spyOn(ScdlCli.prototype, "parseXls").mockResolvedValue();
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+        if (fs.existsSync(testFilePath)) {
+            fs.unlinkSync(testFilePath);
+        }
+    });
+
+    describe("loadConfig method", () => {
+        it("should load config file successfully", () => {
+            fs.writeFileSync(testFilePath, JSON.stringify(validConfigData));
+            const result = loadConfig();
+
+            expect(result).toEqual(validConfigData);
+        });
+
+        it("should throw an error when the JSON file is invalid", () => {
+            const invalidJsonData = "{ invalidJson: true ";
+
+            fs.writeFileSync(testFilePath, invalidJsonData);
+
+            expect(() => {
+                loadConfig();
+            }).toThrowError(new Error("Unexpected token i in JSON at position 2"));
+        });
+
+        it("should throw an error when the JSON file does not exist", () => {
+            expect(() => {
+                loadConfig();
+            }).toThrowError(
+                expect.objectContaining({
+                    message: expect.stringContaining("ENOENT: no such file or directory"),
+                }),
+            );
+        });
+    });
+
+    describe("Test for processFile method", () => {
+        const fileConfig = {
+            name: "donnees-a-integrer1.csv",
+            parseParams: ["producerSlug1", "2025-01-13"] as ScdlParseArgs,
+            addProducer: true,
+            producerName: "Test Producer 1",
+            producerSiret: "12345678901",
+        };
+
+        const fileConfigWrongType = {
+            name: "donnees-a-integrer1.doc",
+            parseParams: ["producerSlug1", "2025-01-13"] as ScdlParseXlsArgs,
+            addProducer: true,
+            producerName: "Test Producer 1",
+            producerSiret: "12345678901",
+        };
+
+        it("should process file methods correctly", async () => {
+            await expect(processFile(fileConfig)).resolves.toBeUndefined();
+
+            expect(addProducerMock).toHaveBeenCalledTimes(1);
+            expect(addProducerMock).toHaveBeenCalledWith(
+                fileConfig.parseParams[0],
+                fileConfig.producerName,
+                fileConfig.producerSiret,
+            );
+
+            expect(parseMock).toHaveBeenCalledTimes(1);
+            expect(parseMock).toHaveBeenCalledWith(
+                expect.stringContaining(fileConfig.name),
+                fileConfig.parseParams[0],
+                fileConfig.parseParams[1],
+                undefined,
+                undefined,
+            );
+            expect(parseXlsMock).not.toHaveBeenCalled();
+        });
+
+        it("should catch error and not call parse method", async () => {
```

Comment:
idem pour le reste, un `it()` pour sous méthode appelée pour bien maîtrisé le code et le retour d'erreur